src/linear_samplers/block_row_gaussian.jl

```diff
@@ -0,0 +1,49 @@
+# This code was Written by Nathaniel Pritchard
+"""
+    LinSysBlkRowGaussSampler <: LinSysBlkRowSampler
+
+A mutable structure with fields to handle Guassian row sketching. 
+
+# Fields
+- `block_size::Int64` - Specifies the size of each block.
+- `sketch_matrix::Union{AbstractMatrix, Nothing}` - The buffer for storing the Gaussian sketching matrix.
+- `scaling::Float64` - The variance of the sketch, is set to be s/n.
+
+Calling `LinSysBlkRowGaussSampler()` defaults to setting `block_size` to 2.
+"""
+mutable struct LinSysBlkRowGaussSampler <: LinSysBlkRowSampler
+    block_size::Int64
+    sketch_matrix::Union{AbstractMatrix, Nothing}
+    scaling::Float64
+end
+
+LinSysBlkRowGaussSampler(block_size) = LinSysBlkRowGaussSampler(block_size, nothing, 0.0)
+LinSysBlkRowGaussSampler() = LinSysBlkRowGaussSampler(2, nothing, 0.0)
+
+# Common sample interface for linear systems
+function sample(
+    type::LinSysBlkRowGaussSampler,
+    A::AbstractArray,
+    b::AbstractVector,
+    x::AbstractVector,
+    iter::Int64
+)
+    m, n = size(A)
+    if iter == 1
+        @assert type.block_size <= m "Block size must be less than row dimension"
+        # Compute scaling for matrix
```

Comment:
I did have that comment later on in the code. I have added a similar comment about how the value is designed so that the scaling is chosen to make the expectation identity